Components/Modal/modal.js

```diff
@@ -0,0 +1,25 @@
+const modalToggleBtn = document.querySelector(".modal-btn");
+const modal = document.querySelector(".modal");
+const closeBtn = document.querySelector(".btn-close");
+const modalOverlayBg = document.querySelector(".modal-overlay");
+
+modalToggleBtn.addEventListener("click", () => {
+    if(!modal.classList.contains("show")) {
```

Comment:
Instead of writing this if-else condition, you can use the classList.toggle. It's quite helpful in these situations.

Reply:
Yup, I did that but realized that I need to do some other DOM manipulations as well within this condition hence cannot use the `.toggle()` method.